### DISCUSSION TITLE
Signed version of older SWT libraries

### DISCUSSION THREAD

**Discussion by chibicitiberiu** [eclipse-platform/eclipse.platform.swt#300] [Q&A] [answered]
I am working on maintaining an older product, which is currently based on Java 8, and we are still distributing 32-bit versions of our software (we have many legacy customers). Some parts of our product use the SWT libraries. Ideally, we would be on the latest version of the Eclipse SDK, but unfortunately this is not possible at this time because we still need the 32-bit libs, and we rely on Java 8.

From my research, the Eclipse SDK no longer distributes 32-bit versions of the SWT libraries, with the last one available being 4.9. Similarly, 4.16 is the last version to support Java 8.

Our security department's main concern is that the SWT DLLs in version 4.9 are not signed binaries, while more recent versions (like 4.16) have signed binaries.

Is there any way to obtain signed DLLs for the older 4.9 version?

**Comment by laeubi** [comment, marked as answer]
> Is there any way to obtain signed DLLs for the older 4.9 version?

You can buy a code-signing certificate and sign the code by yourself and then ship such signed jars/dlls with your product.

**Reply by chibicitiberiu** [reply]
Unfortunately, signing third party binaries is in a bit of a legal grey area, and the company I work for will not assume any of the legal implications of doing that, which is the main reason why I am asking.

**Reply by laeubi** [reply]
> Unfortunately, signing third party binaries is in a bit of a legal grey area

I'm not a laywer so I can't tell much about this

> the company I work for will not assume any of the legal implications of doing that

Then your company should enter a contract with Eclipse that includes maintaining old releases and providing patches to get a new release of SWT out with "official" Eclipse signatures, but this could be a lot of work and effort, you most probably has to maintain and release all by yourself and get less support from the community.

**Reply by chibicitiberiu** [reply]
I was hoping there was a easy way to get those binaries signed, but you're right, it's probably not worth the effort. Our best option at the moment is dropping 32-bit support and going to 4.16. Thank you for your help.
